store/bigtable/config.go

```diff
@@ -0,0 +1,66 @@
+package bigtable
+
+import (
+	"flag"
+	"time"
+
+	"github.com/rakyll/globalconf"
+)
+
+type StoreConfig struct {
+	Enabled           bool
+	GcpProject        string
+	BigtableInstance  string
+	TableName         string
+	WriteQueueSize    int
+	WriteMaxFlushSize int
+	WriteConcurrency  int
+	ReadConcurrency   int
+	MaxChunkSpan      time.Duration
+	ReadTimeout       time.Duration
+	WriteTimeout      time.Duration
+	CreateCF          bool
+}
+
+// return StoreConfig with default values set.
+func NewStoreConfig() *StoreConfig {
+	return &StoreConfig{
+		Enabled:           false,
+		GcpProject:        "default",
+		BigtableInstance:  "default",
+		TableName:         "metrics",
+		WriteQueueSize:    100000,
+		WriteMaxFlushSize: 10000,
+		WriteConcurrency:  10,
+		ReadConcurrency:   20,
+		MaxChunkSpan:      time.Hour * 6,
+		ReadTimeout:       time.Second * 5,
+		WriteTimeout:      time.Second * 5,
+		CreateCF:          true,
+	}
+}
+
+var CliConfig = NewStoreConfig()
+
+func ConfigSetup() {
+	btStore := flag.NewFlagSet("bigtable-store", flag.ExitOnError)
+	btStore.BoolVar(&CliConfig.Enabled, "enabled", CliConfig.Enabled, "enable the bigtable backend store plugin")
+	btStore.StringVar(&CliConfig.GcpProject, "gcp-project", CliConfig.GcpProject, "Name of GCP project the bigtable cluster resides in")
+	btStore.StringVar(&CliConfig.BigtableInstance, "bigtable-instance", CliConfig.BigtableInstance, "Name of bigtable instance")
+	btStore.StringVar(&CliConfig.TableName, "table-name", CliConfig.TableName, "Name of bigtable table used for chunks")
+	btStore.IntVar(&CliConfig.WriteQueueSize, "write-queue-size", CliConfig.WriteQueueSize, "Max number of chunks, per write thread, allowed to be unwritten to bigtable. Must be larger then write-max-flush-size")
+	btStore.IntVar(&CliConfig.WriteMaxFlushSize, "write-max-flush-size", CliConfig.WriteMaxFlushSize, "Max number of chunks in each batch write to bigtable")
+	btStore.IntVar(&CliConfig.WriteConcurrency, "write-concurrency", CliConfig.WriteConcurrency, "Number of writer threads to use")
+	btStore.IntVar(&CliConfig.ReadConcurrency, "read-concurrency", CliConfig.ReadConcurrency, "Number concurrent reads that can be processed")
+	btStore.DurationVar(&CliConfig.MaxChunkSpan, "max-chunkspan", CliConfig.MaxChunkSpan, "Maximum chunkspan size used.")
```

Comment:
yet another config flag for the operator to worry about that can wreak havoc when certain config updates are performed.
i propose we just pass this value in via the constructor, derived from the schemas used.

Reply:
we cant use the max chunkspan of the current schemas, as it may not be the largest chunkspan ever used.  But we can certainly add a validation step to make sure that max-chunkspan is at least as large as the current max chunkspan defined in the schema.